File: KuantumKaos/csharp/kuantumKaos/KaranlikMadde.cs
using System;

namespace kuantumKaos
{
    public class KaranlikMadde : KuantumNesnesi, IKritik
    {
        public override void AnalizEt()
        {
            Stabilite -= 15;

            if (Stabilite <= 0)
            {
                throw new KuantumCokusuException(ID);
            }
        }

        public void AcilDurumSogutmasi()
        {
            Stabilite += 50;

            if (Stabilite > 100)
                Stabilite = 100;

            Console.WriteLine("Karanlık madde soğutuldu.");
        }
    }
}

File: KuantumKaos/python/python_kaos/karanlik_madde.py
from kuantum_nesnesi import KuantumNesnesi
from kritik import Kritik
from kuantum_cokusu_exception import KuantumCokusuException

class KaranlikMadde(KuantumNesnesi, Kritik):
    def analiz_et(self):
        self.stabilite -= 20
        print("Karanlık madde dalgalanıyor!")

        if self.stabilite <= 0:
            raise KuantumCokusuException(self.id)

    def acil_durum_sogutmasi(self):
        self.stabilite += 30
        print("Karanlık madde soğutuldu.")
